src/components/NewEventForm/NewEventForm.tsx

```diff
@@ -0,0 +1,218 @@
+  imageUrl?: string;
+  imageAlt?: string;
+  location?: string;
+  eventType: string;
+  eventStatus: string;
+  teamsUrl?: string;
+};
+
+type EventFormProps = {
+  headerTitle: string;
+  submitTitle: string;
+  onSubmit: (data: EventDataOutput) => Promise<void>;
+} & EventDataInput;
+
+function NewEventForm({ onSubmit, ...rest }: EventFormProps) {
+
+  // Declartions
+  const now = new Date();
+  const tomorrow = addDays(startOfDay(now), 1);
+
+  // Utilities
+  const setHours = (date: Date, hours: number) =>
+    set(date, { hours: hours, minutes: 0, seconds: 0, milliseconds: 0 });
+  const parseDate = (date: string, time: string) => new Date(date + 'T' + time);
+
+  // Functions
+  const eventDate = format(tomorrow, 'yyyy-MM-dd');
+  const eventStartTime = format(setHours(tomorrow, 16), 'HH:mm');
+  const eventEndTime = format(setHours(tomorrow, 17), 'HH:mm');
+
+
+  // Initiation of values
+  const InitialValues = { 
+    name: rest.name || "",
+    eventType: rest.eventType || "",
+    date: eventDate,
+    startTime: eventStartTime,
+    endTime: eventEndTime,
+    description: rest.description || "",
+    location: rest.location || "", 
+    teamsUrl: rest.teamsUrl || "",
+    imageUrl: rest.imageUrl || "",
+    imageAlt: rest.imageAlt || "",
+    eventStatus: rest.eventStatus || EVENT_STATUS_OPTIONS.Draft
+  };
+  
+  // Validation
+  const EventFormSchema = Yup.object().shape({
+    name: Yup
+      .string()
+      .max(100, "Event name is too long")
+      .required("Event name is required"),
+    eventType: Yup
+      .string()
+      .oneOf(EVENT_OPTIONS)
+      .required('Event type is required'),
+    date: Yup
+      .date()
+      .required('Event date is required'),
+    startTime: Yup
+      .string()
+      .required("Start time is required")
+      .test("Start Time", "Start time cannot be later than end time", function(startTime) {
+        const startTimeHour = parseInt(this.parent.startTime.toString().substring(0,2));
+        const endTimeHour = parseInt(this.parent.endTime.toString().substring(0,2));
+        return startTimeHour < endTimeHour;
+      }),
+    endTime: Yup
+      .string()
+      .required("End time is required")
+      /*
+      .test("End Time", "End time cannot be earlier than start time", function(endTime) {
+        const startTimeHour = parseInt(this.parent.startTime.toString().substring(0,2));
+        const endTimeHour = parseInt(this.parent.endTime.toString().substring(0,2));
+        return endTimeHour > startTimeHour;
+      }) */,
+    description: Yup
+      .string()
+      .max(5000, "Description text is too long")
+      .required('Description of event is required'),
+    location: Yup
+      .string()
+      .max(200, "Location text is too long"),
+    imageUrl: Yup
+      .string()
+      .url("URL not valid"),
+    eventStatus: Yup
+      .string()
+      .oneOf([EVENT_STATUS_OPTIONS.Draft, EVENT_STATUS_OPTIONS.Published])
+      .required('Event status is required')
+  });
```

Comment:
This is a constant value, as such, it shouldn't live inside of the component (will cause unnecessary computation on each render)